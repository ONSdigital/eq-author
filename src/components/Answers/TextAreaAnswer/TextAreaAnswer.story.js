import React from "react";
import { storiesOf } from "@storybook/react";
import TextAreaAnswer from "components/Answers/TextAreaAnswer";
import styled from "styled-components";
import { action } from "@storybook/addon-actions";

const Background = styled.div`
  padding: 1em;
  display: block;
  max-width: 20em;
`;

storiesOf("AnswerTypes/TextAreaAnswer", module)
  .addDecorator(story => <Background>{story()}</Background>)
  .add("Empty", () => (
    <TextAreaAnswer
      answer={{ id: "1", label: "", description: "" }}
      onChange={action("changed")}
      onUpdate={action("updated")}
    />
  ))
  .add("Prefilled", () => (
    <TextAreaAnswer
      answer={{
        id: "1",
        label: "Lorem ipsum",
        description: "Nullam id dolor id nibh ultricies."
      }}
      onChange={action("changed")}
      onUpdate={action("updated")}
    />
  ));
