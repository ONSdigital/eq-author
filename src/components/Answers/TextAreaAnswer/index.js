import React from "react";
import PropTypes from "prop-types";
import CustomPropTypes from "custom-prop-types";
import { Field, TextArea } from "components/Forms";
import SeamlessInput from "components/SeamlessInput/SeamlessInput";
import SeamlessTextArea from "components/SeamlessTextArea/SeamlessTextArea";

const TextAreaAnswer = ({ answer, answerIndex, onChange }) =>
  <div>
    <Field id={`answers[${answerIndex}].label`}>
      <SeamlessInput
        placeholder="Label"
        size="medium"
        onChange={onChange}
        value={answer.label}
        data-autoFocus
      />
    </Field>
    <Field id={`answers[${answerIndex}].description`}>
      <SeamlessTextArea
        cols="30"
        rows="5"
        placeholder="Enter a description (optional)â€¦"
        onChange={onChange}
        value={answer.description}
      />
    </Field>
    <div>
      <TextArea rows={5} disabled />
    </div>
  </div>;

TextAreaAnswer.propTypes = {
  answer: CustomPropTypes.answer.isRequired,
  onChange: PropTypes.func.isRequired,
  answerIndex: PropTypes.number.isRequired
};

export default TextAreaAnswer;
