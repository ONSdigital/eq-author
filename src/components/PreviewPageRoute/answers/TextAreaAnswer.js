import React from "react";

import { Field, Label, Input } from "./elements";

const Textarea = Input.withComponent("textarea");

export default ({ answer }) => (
  <Field>
    <Label description={answer.description}>{answer.label}</Label>
    <Textarea cols="60" rows="8" />
  </Field>
);
