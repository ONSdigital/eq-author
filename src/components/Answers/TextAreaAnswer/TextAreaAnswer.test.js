import React from "react";
import { shallow } from "enzyme";
import TextAreaAnswer from "components/Answers/TextAreaAnswer";
import createMockStore from "tests/utils/createMockStore";

const answer = {
  title: "Lorem ipsum",
  description: "Nullam id dolor id nibh ultricies."
};

describe("TextAreaAnswer", () => {
  let handleChange;
  let handleUpdate;
  let component;
  let store;

  beforeEach(() => {
    handleChange = jest.fn();
    handleUpdate = jest.fn();
    store = createMockStore();
    component = shallow(
      <TextAreaAnswer
        onChange={handleChange}
        onUpdate={handleUpdate}
        answer={answer}
        store={store}
      />
    );
  });

  it("should render", () => {
    expect(component).toMatchSnapshot();
  });
});
