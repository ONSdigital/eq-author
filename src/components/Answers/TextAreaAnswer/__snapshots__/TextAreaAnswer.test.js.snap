// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextAreaAnswer should render 1`] = `
<div>
  <Field
    id="answers[1].label"
  >
    <withSeamlessness(withChangeHandler(Input))
      data-autoFocus={true}
      onChange={[Function]}
      placeholder="Label"
      size="medium"
    />
  </Field>
  <Field
    id="answers[1].description"
  >
    <withSeamlessness(withChangeHandler(SeamlessTextArea__TextArea))
      cols="30"
      onChange={[Function]}
      placeholder="Enter a description (optional)â€¦"
      rows="5"
      size="small"
      value="Nullam id dolor id nibh ultricies."
    />
  </Field>
  <div>
    <withChangeHandler(TextArea)
      disabled={true}
      rows={5}
    />
  </div>
</div>
`;
